### PR TITLE
add user_id and team_id as log facets

### PR DESCRIPTION
**Pull request by song-william**
Add `user_id` and `team_id` as additional log facets. Clean up the log tagging to be a bit more general. 

Tested `CustomJSONFormatter` locally and saw that request_id, user_id, and team_id are in the logger extras.

### CODE REVIEW

File: model-engine/model_engine_server/core/loggers.py [hunk 1]
```diff
@@ -77,10 +92,26 @@ def json_record(self, message: str, extra: dict, record: logging.LogRecord) -> d
         extra["lineno"] = record.lineno
         extra["pathname"] = record.pathname
 
-        # add the http request id if it exists
-        request_id = ctx_var_request_id.get()
-        if request_id:
-            extra["request_id"] = request_id
+        # # add the http request id if it exists
```
Comment (seanshi-scale) [inline model-engine/model_engine_server/core/loggers.py:95]: remove commented out stuff?

Comment (yixu34) [inline model-engine/model_engine_server/core/loggers.py:95]: Delete these commented out lines?

File: model-engine/model_engine_server/api/app.py [hunk 2]
```diff
@@ -23,9 +23,8 @@
 from model_engine_server.api.triggers_v1 import trigger_router_v1
 from model_engine_server.core.loggers import (
     filename_wo_ext,
-    get_request_id,
     make_logger,
-    set_request_id,
+    LoggerTagManager, LoggerTagKey
```
Comment (yixu34) [inline model-engine/model_engine_server/api/app.py:27]: Does this pass the linter?

**Review by yixu34 — APPROVED** [review]

**Review by ian-scale — APPROVED** [review]
Looks good to me after you remove the commented lines!

File: model-engine/model_engine_server/core/loggers.py [hunk 3]
```diff
@@ -35,19 +34,35 @@
...
-def set_request_id(request_id: str) -> None:
-    """Set the request id in the context variable."""
-    ctx_var_request_id.set(request_id)  # type: ignore
+class LoggerTagKey(str, Enum):
+    REQUEST_ID = 'request_id'
+    TEAM_ID = 'team_id'
+    USER_ID = 'user_id'
+
+class LoggerTagManager:
+    _context_vars: Dict[LoggerTagKey, contextvars.ContextVar] = {}
```
Comment (seanshi-scale) [inline model-engine/model_engine_server/core/loggers.py:47]: wonder if a `defaultdict` works here, although I guess it's probably more effort to get the code into a form where it works with `defaultdict`

**Review by seanshi-scale — APPROVED** [review]
lgtm once the commented out code is removed (if we want to remove it)

File: model-engine/model_engine_server/api/app.py (hunk 2 above)
Comment (song-william) [reply model-engine/model_engine_server/api/app.py:27]: Probably not, my pre-commit setup is a little wonky with submodules right now.

**Pull request merged**